client/ayon_core/plugins/publish/collect_hierarchy.py

```diff
@@ -49,19 +48,32 @@ def process(self, context):
                 "folder_type": "Shot",
                 "tasks": instance.data.get("tasks") or {},
                 "comments": instance.data.get("comments", []),
-                "attributes": {
-                    "handleStart": instance.data["handleStart"],
-                    "handleEnd": instance.data["handleEnd"],
-                    "frameStart": instance.data["frameStart"],
-                    "frameEnd": instance.data["frameEnd"],
-                    "clipIn": instance.data["clipIn"],
-                    "clipOut": instance.data["clipOut"],
-                    "fps": instance.data["fps"],
-                    "resolutionWidth": instance.data["resolutionWidth"],
-                    "resolutionHeight": instance.data["resolutionHeight"],
-                    "pixelAspect": instance.data["pixelAspect"],
-                },
             }
+
+            shot_data["attributes"] =  {}
+            SHOT_ATTRS = (
+                "handleStart",
+                "handleEnd",
+                "frameStart",
+                "frameEnd",
+                "clipIn",
+                "clipOut",
+                "fps",
+                "resolutionWidth",
+                "resolutionHeight",
+                "pixelAspect",
+            )
+            for shot_attr in SHOT_ATTRS:
+                if shot_attr not in instance.data:
+                    # Shot attribute might not be defined (e.g. CSV ingest)
+                    self.log.debug(
+                        "%s shot attribute is not defined for instance.",
+                        shot_attr
+                    )
+                    continue
+
+                shot_data["attributes"][shot_attr] = instance.data[shot_attr]
```

Comment:
Maybe we should skip even if the key is filled with `None`?
```suggestion
                attr_value = instance.data.get(shot_attr)
                if attr_value is None:                
                    # Shot attribute might not be defined (e.g. CSV ingest)
                    self.log.debug(
                        "%s shot attribute is not defined for instance.",
                        shot_attr
                    )
                    continue

                shot_data["attributes"][shot_attr] = attr_value
```